### ISSUE TITLE
Typescript!

### ISSUE THREAD

**Issue by alyahmedaly** [javierbrea/eslint-plugin-boundaries#107]
**Is your feature request related to a problem? Please describe.**
I'm always frustrated when try to be the first one to use typescript with `eslint-plugin-boundaries` 😀

**Describe the solution you'd like**
debugging `eslint-plugin-boundaries` I found it's using `const resolve = require("eslint-module-utils/resolve").default;` to resolve import paths which lead me to typescript section https://github.com/benmosher/eslint-plugin-import#typescript

so long story short adding  `- plugin:import/typescript # this line does the trick` to extends works!

**Comment by javierbrea** [comment]
Hi @alyahmedaly,

I'm sorry but I don't understand what do you mean. Did you achieve configuring the plugin to work with TypeScript, or are you facing any issue? Anyway, could you provide an example of the configuration? If it is working, I'd like to add an example to the documentation, so it may be useful. If it is not working, I'd like to help 🙂

**Comment by Amareis** [comment]
I try to use this plugin with my typescript project, and it's definitely not working with TS paths aliases. When I added https://www.npmjs.com/package/eslint-import-resolver-typescript, it starts throwing strange errors: 
![image](https://user-images.githubusercontent.com/4806617/121527216-9d77de00-ca13-11eb-9370-e5a4890ac856.png)

Can you document how to use this plugin with typescript?

**Comment by lindekaer** [comment]
@javierbrea is the module supposed to be able to work on Typescript files?

**Comment by javierbrea** [comment]
Hi @lindekaer,
To be honest, I still haven't tested it, that's why this issue is still opened. I'll try to test it as soon as possible. Anyway, contributions are also welcome, so, comments on this issue from anyone who tries and makes it works would be great 😃

**Comment by lindekaer** [comment]
@javierbrea thanks a lot for your quick response!  🌟
I'm not sure I can help with the contribution, but I'll share my example TS project that I'm unable to get working.

**Comment by javierbrea** [comment]
Hi @lindekaer, @Amareis ,

I have just created a TypeScript project for testing, and I have achieved configuring it properly. I used `@typescript-eslint/eslint-plugin`, `@typescript-eslint/parser` and `eslint-import-resolver-typescript`, and I configured also TS path alisases without problem.

Here you have an example of my `.eslintrc.js` file:

```js
module.exports = {
  root: true,
  env: {
    node: true,
    es6: true,
  },
  parser: "@typescript-eslint/parser",
  plugins: ["@typescript-eslint", "prettier", "boundaries"],
  extends: [
    "eslint:recommended",
    "plugin:@typescript-eslint/recommended",
    "prettier",
    "plugin:boundaries/strict",
  ],
  settings: {
    "boundaries/elements": [
      {
        type: "root",
        pattern: "src/index.ts",
        mode: "full",
      },
      {
        type: "components",
        pattern: "components/*",
      },
      {
        type: "modules",
        pattern: "modules/*",
      },
    ],
    "boundaries/include": ["src/**/*.*"],
    "import/resolver": {
      typescript: {
        alwaysTryTypes: true,
      },
    },
  },
  rules: {
    "prettier/prettier": [
      "error",
      {
        printWidth: 99,
        parser: "flow",
      },
    ],
    "no-undef": "error",
    "no-unused-vars": ["error", { vars: "all", args: "after-used", ignoreRestSiblings: false }],
    "boundaries/element-types": [
      2,
      {
        default: "disallow",
        rules: [
          {
            from: "root",
            allow: ["modules"],
          },
          {
            from: "components",
            allow: ["components"],
          },
          {
            from: "modules",
            allow: ["components"],
          },
        ],
      },
    ],
  },
};
```

I will add configuration instructions to the documentation, and I will also create a repository with the example I have just created.
Please provide some examples of your configuration in order to find out why it is not working in your cases.

**Comment by javierbrea** [comment]
I have added [documentation about usage with TypeScript](https://github.com/javierbrea/eslint-plugin-boundaries#usage-with-typescript), and I have created [this repository with a fully working and tested example](https://github.com/javierbrea/epb-ts-example).

Please let me know if following the docs do you still face any issue.

TypeScript docs was a good idea, thanks to all for your contribution! 😃
